{
  "url": "https://api.github.com/repos/shap/shap/issues/1621",
  "repository_url": "https://api.github.com/repos/shap/shap",
  "labels_url": "https://api.github.com/repos/shap/shap/issues/1621/labels{/name}",
  "comments_url": "https://api.github.com/repos/shap/shap/issues/1621/comments",
  "events_url": "https://api.github.com/repos/shap/shap/issues/1621/events",
  "html_url": "https://github.com/shap/shap/issues/1621",
  "id": 756447626,
  "node_id": "MDU6SXNzdWU3NTY0NDc2MjY=",
  "number": 1621,
  "title": "Several bug fix to call SHAP from EconML",
  "user": {
    "login": "heimengqi",
    "id": 27828915,
    "node_id": "MDQ6VXNlcjI3ODI4OTE1",
    "avatar_url": "https://avatars.githubusercontent.com/u/27828915?v=4",
    "gravatar_id": "",
    "url": "https://api.github.com/users/heimengqi",
    "html_url": "https://github.com/heimengqi",
    "followers_url": "https://api.github.com/users/heimengqi/followers",
    "following_url": "https://api.github.com/users/heimengqi/following{/other_user}",
    "gists_url": "https://api.github.com/users/heimengqi/gists{/gist_id}",
    "starred_url": "https://api.github.com/users/heimengqi/starred{/owner}{/repo}",
    "subscriptions_url": "https://api.github.com/users/heimengqi/subscriptions",
    "organizations_url": "https://api.github.com/users/heimengqi/orgs",
    "repos_url": "https://api.github.com/users/heimengqi/repos",
    "events_url": "https://api.github.com/users/heimengqi/events{/privacy}",
    "received_events_url": "https://api.github.com/users/heimengqi/received_events",
    "type": "User",
    "site_admin": false
  },
  "labels": [

  ],
  "state": "open",
  "locked": false,
  "assignee": null,
  "assignees": [

  ],
  "milestone": null,
  "comments": 3,
  "created_at": "2020-12-03T18:33:20Z",
  "updated_at": "2020-12-04T21:43:56Z",
  "closed_at": null,
  "author_association": "CONTRIBUTOR",
  "active_lock_reason": null,
  "body": "Hi, I created a PR solving issues I found when implemented SHAP functionalities into EconML #1620 , please take a look and we reply on those changes to make it work calling from EconML. \r\n\r\nAlso I have another issue didn't solve through this PR, when I have a Explanation object and I reset the `feature_names` as a list, it can't be correctly passed through when call slicing, the way to make it work correctly is to reset as an alias of list `Alias(list of new feature_names, dimension)`, could you please handle that internally?\r\n\r\n```\r\nest=LinearRegression()\r\nest.fit(x,y)\r\nexplainer=shap.Explainer(est,x[:10],feature_names=[\"a\",\"b\",\"c\",\"d\"])\r\nshap_values=explainer(x[:10])\r\n# change feature names\r\nshap_values.feature_names=[\"a\",\"b\",\"c\"]\r\n# new feature name won't pass through when slicing\r\nprint(shap_values.feature_names)\r\nprint(shap_values[1].feature_names)\r\n```\r\n\r\n ",
  "closed_by": null,
  "reactions": {
    "url": "https://api.github.com/repos/shap/shap/issues/1621/reactions",
    "total_count": 0,
    "+1": 0,
    "-1": 0,
    "laugh": 0,
    "hooray": 0,
    "confused": 0,
    "heart": 0,
    "rocket": 0,
    "eyes": 0
  },
  "timeline_url": "https://api.github.com/repos/shap/shap/issues/1621/timeline",
  "performed_via_github_app": null,
  "state_reason": null
}
